ci-operator/step-registry/ibmcloud/provision/vpc/ibmcloud-provision-vpc-commands.sh

```diff
@@ -0,0 +1,201 @@
+
+    waitAvailable "vpc" ${vpcName}
+    
+    # create subnets
+    zones=("${region}-1" "${region}-2" "${region}-3")
+
+    for zone in "${zones[@]}"; do
+        zone_cidr=$(getZoneAddressprefix "${vpcName}" "${zone}")
+        zone_cidr_main="${zone_cidr%/*}"
+        subnetName="${preName}-control-plane-${zone}"
+        "${IBMCLOUD_CLI}" is subnet-create ${subnetName} ${vpcName} --ipv4-cidr-block "${zone_cidr_main}/24" || return 1
+        waitAvailable "subnet" ${subnetName}
+    done
+
+    for zone in "${zones[@]}"; do
+        zone_cidr=$(getZoneAddressprefix "${vpcName}" "${zone}")
+        zone_cidr_main=$(getOneMoreCidr "${zone_cidr%/*}")
+        subnetName="${preName}-compute-${zone}"
+        "${IBMCLOUD_CLI}" is subnet-create ${subnetName} ${vpcName} --ipv4-cidr-block "${zone_cidr_main}/24" || return 1
+        waitAvailable "subnet" ${subnetName}
+    done
+}
+
+function check_vpc() {
+    local vpcName="$1" vpc_info_file="$2"
+
+    "${IBMCLOUD_CLI}" is vpc ${vpcName} --show-attached --output JSON > "${vpc_info_file}" || return 1
+}
+
+function string2arr() {
+    # parameter:
+    # $1: a string with seperator. Note: each item by seperator can not include whitespace
+    # $2: seperator, by default, it is ,
+    local intput_string="$1" seperator="$2" output_array
+
+    if [[ -z "$seperator" ]]; then
+        seperator=","
+    fi
+
+    IFS="$seperator" read -r -a output_array <<< "${intput_string}"
+    echo "${output_array[@]}"
+}
+
+function getAddressPre() {
+    local vpc_info_file="$1"
+    local ip ips
+    ip=$(cat ${vpc_info_file} | jq -c -r .address_prefixes[0].cidr)
+    IFS="." read -ra ips <<< "${ip}"
+    echo "${ips[0]}.${ips[1]}.0.0/16"
+}
+
+function create_zone_public_gateway() {
+    local gateName="$1" vpcName="$2" zone="$3"
+
+    "${IBMCLOUD_CLI}" is public-gateway-create ${gateName} ${vpcName} ${zone} || return 1
+}
+
+function attach_public_gateway_to_subnet() {
+    local subnetName="$1" vpcName="$2" pgwName="$3"
+
+    "${IBMCLOUD_CLI}" is subnet-update ${subnetName} --vpc ${vpcName} --pgw ${pgwName} || return 1
+}
+
+checkCli
+
+
+ibmcloud_login || exit 1
+
+## Create the VPC
+echo "$(date -u --rfc-3339=seconds) - Creating the VPC..."
+
+CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+region="${LEASED_RESOURCE}"
+
+resource_group="${CLUSTER_NAME}-rg"
+
+create_resource_group ${resource_group}
+echo "${resource_group}" > "${SHARED_DIR}/ibmcloud_resource_group"
+
+echo "Create VPC..."
+vpc_name="${CLUSTER_NAME}-vpc"
+echo "${vpc_name}" > "${SHARED_DIR}/ibmcloud_vpc_name"
+create_vpc "${CLUSTER_NAME}" "${vpc_name}" "${resource_group}" "${region}" || exit 3
+
+vpc_info_file="${ARTIFACT_DIR}/vpc_info"
+check_vpc "${vpc_name}" "${vpc_info_file}" || exit 3
+
+controlPlaneSubnets=$(cat "${vpc_info_file}" | jq -c -r '[.subnets[] | select(.name|test("control-plane")) | .name]')
+IFS="," read -r -a controlPlaneSubnetAry <<< ${controlPlaneSubnets:1:-1}
```

Comment:
Actually no need the complex way to convert string, more details in https://github.com/openshift/release/pull/34116/files#r1040449314